provider/file/file_test.go

```diff
@@ -0,0 +1,167 @@
+// Copyright © 2023 Bank-Vaults Maintainers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package file
+
+import (
+	"context"
+	"os"
+	"strings"
+	"testing"
+)
+
+func TestMain(m *testing.M) {
+	exitCode := m.Run()
+
+	// teardown environment variables after tests are done
+	teardownEnvs()
+
+	os.Exit(exitCode)
+}
+
+func TestNewFileProvider(t *testing.T) {
+	// create a new secret file and write secrets into it
+	tmpfile := createTempFileWithContent(t)
+	defer os.Remove(tmpfile.Name())
+
+	// create new environment variables
+	setupEnvs(t, tmpfile)
+
+	fileProvider, err := NewFileProvider(os.Getenv("SECRETS_FILE_PATH"))
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
We use https://pkg.go.dev/gotest.tools/v3/assert package for testing.
The error checking becomes a one-liner.